### ISSUE TITLE
Chapter 10: Evaluating `(ext1 rectify-0 0)` causes a contract failure

### ISSUE THREAD

**Issue by jmikkola** [themetaschemer/malt#21]
On page 197 of Chapter 10, it lists code to define `rectify`:

```racket
(define rectify
  (ext1 rectify-0 0))
```

Evaluating this fails with

```
; <: contract violation
;   expected: real?
;   given: #<procedure:ρ-function>
; Context (plain; to see better errortrace context, re-run with C-u prefix):
;    <my home dir>/Projects/little-learner/little-learner.rkt:465:2 rectify-0
;    <my home dir>.racket/8.6/pkgs/malt/flat-tensors/autodiff/B-prims.rkt:207:2 ext1
;    <my home dir>/.emacs.d/elpa/racket-mode-20230209.1424/racket/repl.rkt:348:0 configure/require/enter
```

I've tried running `raco pkg update malt` today. It installed an update, but didn't fix this issue.

**Comment by themetaschemer** [comment]
Thanks for reporting the issue. Looking into it. As a workaround, please switch to the learner implementation per the instructions in the README or the website.

**Comment by jmikkola** [comment]
Thank you!

If I understood correctly, I should replace `(require malt)` with `(require malt/learner)`.

When I do that, other parts of the code fail because `declare-hyper` is unbound.

I think I can fix that (and a similar issue with `dot-product` being unbound) by adding both `(require malt/tools)` and `(require malt/malted)`. 

However, using those imports causes later issues when running the Iris model (frame 266:71)

```
; length: contract violation
;   expected: list?
;   given: 0.5
; Context (plain; to see better errortrace context, re-run with C-u prefix):
;   /home/jeremy/.racket/8.6/pkgs/malt/flat-tensors/tensors/D-extend.rkt:14:4
;   /home/jeremy/.racket/8.6/pkgs/malt/flat-tensors/autodiff/B-prims.rkt:48:2 prim1-dual
;   /home/jeremy/.racket/8.6/pkgs/malt/malted/O-init.rkt:25:6
;   /home/jeremy/.racket/8.6/pkgs/malt/flat-tensors/tensors/B-tensor-basics.rkt:150:2 fill-flat-tensor
;   /home/jeremy/.racket/8.6/pkgs/malt/flat-tensors/tensors/B-tensor-basics.rkt:150:2 fill-flat-tensor
;   /home/jeremy/.racket/8.6/pkgs/malt/flat-tensors/tensors/B-tensor-basics.rkt:142:2 build-tensor
;   /usr/share/racket/collects/racket/private/map.rkt:40:19 loop
;   /home/jeremy/.racket/8.6/pkgs/malt/tools/A-hypers.rkt:43:13
```

**Comment by themetaschemer** [comment]
Oh, sorry. I just noticed that you're using the default racket pkg location. Can you try the following? 
```
cd /home/jeremy/.racket/8.6/pkgs/malt/
echo "(tensor-implementation learner)" > local.cfg
make clean && make
```
That should switch your default implementation to learner. You can then simply (require malt).

**Comment by themetaschemer** [comment]
@jmikkola Thanks for your patience. I have just pushed a new version of Malt which as `learner` as the default implementation. This will enable you to run all code in the book without issues (known issues, that is :-). If you haven't already followed my instructions above, updating the malt package again should fix it for you. 

Fixed in c8220b5.
